File: paths.cpp
#include <iostream>
#include <cmath>
#include <random>



int main(){

    int i= 1;
    int T=0;
    std::default_random_engine generator;
    std::normal_distribution<double> normal(0,.1); 
    std::cout<<"\nEnter an endpoint for the time interval"<<std::endl;
    std::cin >> T;
    double time[T*100+1];
    double path[T*100+1];
    time[0]=0;
    path[0]=0;
    double *ppath=path;
    for(double *ptime=&time[1], *ptime_end=time+(T*100+1);ptime !=ptime_end; ++ptime and ++ppath){
        *ptime=i*(.01);
        *ppath=*(ppath-1)+normal(generator);
        std::cout   <<i
                    <<':'
                    <<*ptime
                    <<':'
                    <<*ppath
                    << std::endl;
        i=i+1;
    }
    return 0;
}

File: paths.py
import numpy as np  
import pandas as pd
import matplotlib.pyplot as plt

def make_paths(n):
    #make_paths will be called in main.  This function will fill a #dataframe with the values of the brownian motion
	df1=pd.DataFrame(index=range(n), columns=['BM1'])
	df1.loc[0,'BM1']=0
	k=1
	while k<len(df1):
		df1.loc[k,'BM1']=df1.loc[k-1,'BM1']+np.random.normal(0,.1)
		k=k+1
	return(df1)



def main(): 
	
	d=int(input('How many sample paths would you like to see?\n'))
	T=int(input('What is the endpoint of your time interval? Enter a positive integer.\n'))
	
    #Our time increments will be a constant .01, convenient for
    # calculation purposes.  Need to be able to take square 
    # root of increment to get standard deviation for use in 
    # random.normal


    #time interval is [0,T].  There are T*100 increments, and n
    # endpoints
	n=T*100+1
	t=np.linspace(0,T,n)

	i=0
	while i<d:
		G=plt.plot(t, make_paths(n))
		i=i+1
	
	plt.show()

if __name__=='__main__': 
	main()